torchtune/models/llava_next/image_processor.py

```diff
@@ -0,0 +1,344 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+
+import math
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import PIL
+
+import torch
+
+import torchvision
+from torch import nn
+
+from torchtune.utils.image_transforms_utils import find_supported_resolutions
+from torchvision.transforms import v2
+from torchvision.transforms.v2 import functional as F
+
+logger = logging.getLogger(__name__)
+
+ImageInput = Union["PIL.Image.Image", np.ndarray, "torch.Tensor"]
+
+
+class GetImagePatches(nn.Module):
+    def __init__(
+        self,
+        possible_resolutions: List[Tuple[int, int]],
+        patch_size: int,
+        resample: Union[str, torchvision.transforms.InterpolationMode] = "bilinear",
+    ) -> None:
+        super().__init__()
+        """
+        Class used to divide and image with variable resolutions into patches of equal sizes,
+        including the original image resized to [patch_size, patch_size].
+
+        Forward returns a tensor of shape [num_patches + 1, channels, patch_size, patch_size].
+
+        Args:
+            possible_resolutions (List[Tuple[int, int]]): List of possible resolutions as [height, width].
+            patch_size (int): Size of the patches to divide the image into.
+            resample (Union[str, torchvision.transforms.InterpolationMode]):
+            Resampling method, either 'bilinear', 'bicubic' or torchvision.transforms.InterpolationMode.
+        """
+
+        self.possible_resolutions = torch.tensor(possible_resolutions)
+
+        if isinstance(resample, torchvision.transforms.InterpolationMode):
+            self.resample = resample
+        if resample == "bilinear":
+            self.resample = torchvision.transforms.InterpolationMode.BILINEAR
+        elif resample == "bicubic":
+            self.resample = torchvision.transforms.InterpolationMode.BICUBIC
+        else:
+            raise ValueError(
+                "resample must be of type torchvision.transforms.InterpolationMode or ['bilinear', 'bicubic]."
+            )
```

Comment:
I was thinking that it could make it more convenient for the user. Instead of having to another library just to define the interpolation, for example, it can just be a string. I am fine with removing it though.

Reply:
Another option is to keep it as string only, then access the enum with the string:
```
In [5]: InterpolationMode["BILINEAR"]
Out[5]: <InterpolationMode.BILINEAR: 'bilinear'>
```
that way you don't need to maintain these if statements